domain/test/usecases/shared_space/create_shared_space_folder_interactor_test.dart

```diff
@@ -0,0 +1,104 @@
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:testshared/fixture/shared_space_document_fixture.dart';
+import 'package:data/data.dart';
+
+import '../../mock/repository/mock_shared_space_document_repository.dart';
+
+void main() {
+  group('create_shared_space_folder_interactor test', () {
+    MockSharedSpaceDocumentRepository sharedSpaceDocumentRepository;
+    CreateSharedSpaceFolderInteractor createSharedSpaceFolderInteractor;
+    final request = CreateSharedSpaceNodeFolderRequest('Dat is good', sharedSpaceFolder1.workGroupNodeId);
+
+    setUp(() {
+      sharedSpaceDocumentRepository = MockSharedSpaceDocumentRepository();
+      createSharedSpaceFolderInteractor = CreateSharedSpaceFolderInteractor(sharedSpaceDocumentRepository);
+    });
+
+    test('Create Shared Space Folder should return success with valid data', () async {
+      when(sharedSpaceDocumentRepository.createSharedSpaceFolder(
+        sharedSpaceFolder1.sharedSpaceId,
+        request
+      )).thenAnswer((_) async => workGroupFolder1);
+
+      final result = await createSharedSpaceFolderInteractor.execute(
+        sharedSpaceFolder1.sharedSpaceId,
+        request
+      );
+
+      final workGroups = result
+          .map((success) => (success as CreateSharedSpaceFolderViewState).workGroupFolder)
+          .getOrElse(() => null);
+      expect(workGroups, workGroupFolder1);
```

Comment:
please verify the `parentId` created folder also.

Reply:
Can you be more specific please what do I need to test ?

Reply:
```
expect(createdFolder.parentId, request.parentWorkGroupNodeId)
```